stake-pool/cli/src/main.rs

```diff
@@ -143,299 +230,343 @@ fn command_create_account(config: &Config, token: Pubkey) -> CommandResult {
-        .get_token_accounts_by_owner_with_commitment(
-            &config.owner.pubkey(),
-            match token {
-                Some(token) => TokenAccountsFilter::Mint(token),
-                None => TokenAccountsFilter::ProgramId(spl_token::id()),
-            },
-            config.commitment_config,
-        )?
-        .value;
+    let mut accounts = get_authority_accounts(config, &pool_withdraw_authority);
     if accounts.is_empty() {
-        println!("None");
+        println!("No accounts found.");
+        return Ok(None);
     }
+    // Sort from lowest to highest balance
+    accounts.sort_by(|a, b| a.1.lamports.cmp(&b.1.lamports));
 
-    println!("Account                                      Token                                        Balance");
-    println!("-------------------------------------------------------------------------------------------------");
-    for keyed_account in accounts {
-        let address = keyed_account.pubkey;
-
-        if let UiAccountData::Json(parsed_account) = keyed_account.account.data {
-            if parsed_account.program != "spl-token" {
-                println!(
-                    "{:<44} Unsupported account program: {}",
-                    address, parsed_account.program
-                );
-            } else {
-                match serde_json::from_value(parsed_account.parsed) {
-                    Ok(TokenAccountType::Account(ui_token_account)) => println!(
-                        "{:<44} {:<44} {}",
-                        address, ui_token_account.mint, ui_token_account.token_amount.ui_amount
-                    ),
-                    Ok(_) => println!("{:<44} Unsupported token account", address),
-                    Err(err) => println!("{:<44} Account parse failure: {}", address, err),
-                }
-            }
+    for (stake_pubkey, account) in accounts {
+        if account.lamports < amount {
+            continue;
+        }
+        // Account found to claim or withdraw
+        println!("Withdrawing from account {}", stake_pubkey);
+
+        let mut instructions: Vec<Instruction> = vec![];
+        let mut signers = vec![config.fee_payer.as_ref(), config.owner.as_ref()];
+
+        let stake_receiver_account = Keypair::new();
+
+        let mut total_rent_free_balances: u64 = 0;
+
+        // Calculate amount of tokens to burn
+        let tokens_to_burn = (amount as u128)
+            .checked_mul(pool_data.pool_total as u128)
+            .unwrap()
+            .checked_div(pool_data.stake_total as u128)
+            .unwrap() as u64;
+
+        instructions.push(
+            // Approve spending token
+            approve_token(
+                &spl_token::id(),
+                &burn_from,
+                &pool_withdraw_authority,
+                &config.owner.pubkey(),
+                &[],
+                tokens_to_burn,
+            )?,
+        );
+
+        if amount == account.lamports {
+            // Claim to get whole account
```

Comment:
If someone has specified `stake_receiver`, you'll need to add instructions to move the stake to that account.  If there isn't a `stake_receiver`, please print out the account being claimed

Reply:
Any answer here?

Reply:
Currently the logic is to split existing stake directly into stake_receiver. What you're suggesting is to always use a new account to split stake into and then merge into stake_receiver (if specified)?

Reply:
The logic to split is on the else branch, and that's perfect.  On this branch though, the `Claim` instruction will just claim the account without looking at `stake_receiver`.  When someone specifies that they want to withdraw to `stake_receiver`, they don't care if it comes from a `claim` or a `split` internally, they'll just want the stake to go to `stake_receiver`.  So in this case, you'll need to do another instruction to move the stake into `stake_receiver`.  Does that make sense?

Reply:
Got it